### DISCUSSION TITLE
Option for `t.fail()` to not output stacktrace/failure report

### DISCUSSION THREAD

**Discussion by fredboyle** [avajs/ava#2746] [Ideas]
## Trying to

Remove stacktrace/failure report from AVA results ouput.

### Example code

```js
await foo.close().catch(error => {
    const testResults = error.getTestResults();

    if (testResults.getStatus() === 'Passed') {
      t.pass(`👍 No diffs found!`);
    } else {
      t.fail(`⁉️  Test is ${testResults.getStatus()}`);
    }
});
```

### Output

```zsh
» Test should pass

  __tests__/foo.spec.js:XX

   96:         } else {                                           
   97:           t.fail(`⁉️  Test is ${testResults.getStatus()}`);
   98:         }     

  ⁉️  Test is Unresolved
```

### Goal

Suppress the output above. 
I know that when using `t.fail()` the test will fail at that line. 
I only need to know that the test failed, not where in the code in such cases.

**Comment by novemberborn** [comment]
In what way do these extra lines of output make it harder for you to use AVA?

**Reply by fredboyle** [reply]
It creates unnecessary noise which buries the testing results and summary. It increases mental load and reduces the positive developer experience.

**Reply by novemberborn** [reply]
I can see that for your use case. There might be other use cases where having that extra bit of context is useful for understanding where the failure occurred. This isn't something we're interested in making configurable. Thank you for the suggestion though.
